kumascript/src/render.js

```diff
@@ -248,51 +265,23 @@ async function render(source, templates, pageEnvironment, allPagesInfo) {
           // There was a runtime error executing the macro
           macroError = new MacroExecutionError(e, source, token);
         }
-        fatalErrors.push(macroError);
+        currentResult.errors.fatal = macroError;
+        // There was a fatal error while rendering this macro, so
+        // just use the original macro source text for the output.
+        currentResult.output = source.slice(
+          token.location.start.offset,
+          token.location.end.offset
+        );
       }
     }
+    output += currentResult.output;
+    if (currentResult.errors.fatal) {
+      errors.push(currentResult.errors.fatal);
+    } else if (currentResult.errors.nonFatal) {
+      errors.push(...currentResult.errors.nonFatal);
```

Comment:
Why does it do a shallow copy of  the nonFatals and not of the fatals?

Reply:
I'm actually spreading `currentResult.errors.nonFatal` into arguments to the `push`. I originally had:
```js
for (const e of currentResult.errors.nonFatal) {
  errors.push(e);
}
```
which I think is more robust in the sense that the size of `currentResult.errors.nonFatal` doesn't matter compared to what I have (where there's a limit to the number of arguments that the push can handle). On further reflection, I think I'll go back to what I originally had. It's more robust.